include/picongpu/fields/background/templates/twtsfast/BField.tpp

```diff
@@ -650,18 +657,37 @@ namespace picongpu
                 float_T const w0 = float_T(w_x_SI / UNIT_LENGTH);
                 float_T const rho0 = float_T(PI * w0 * w0 / lambda0);
                 float_T const k = float_T(2.0 * PI / lambda0);
-                /* If phi < 0 the entire pulse is rotated by 180 deg around the
-                 * z-axis of the coordinate system without also changing
-                 * the orientation of the resulting field vectors.
+
+                /* In order to calculate in single-precision and in order to account for errors in
+                 * the approximations far from the coordinate origin, we use the wavelength-periodicity and
+                 * the known propagation direction for realizing the laser pulse using relative coordinates
+                 * (i.e. from a finite coordinate range) only. All these quantities have to be calculated
+                 * in double precision.
                  */
+                float_64 sinPhiVal;
+                float_64 cosPhiVal;
+                math::sincos(phi, sinPhiVal, cosPhiVal);
+                float_64 const tanAlpha = (1.0 - beta_0 * cosPhiVal) / (beta_0 * sinPhiVal);
+                float_64 const tanFocalLine = math::tan(PI / float_64(2.0) - phi);
```

Comment:
Please remove `float_64(2.0)` with `2.0`, it is already double precision. We add the precision only if the type has a variable precision or where we need to downcast.

Reply:
Done for all occurrences in the PR.